library/alloc/src/boxed.rs

```diff
@@ -343,6 +343,29 @@ impl<T, A: Allocator> Box<T, A> {
         let (raw, alloc) = Box::into_raw_with_allocator(boxed);
         unsafe { Box::from_raw_in(raw as *mut [T; 1], alloc) }
     }
+
+    /// Consumes the `Box`, returning the wrapped value.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(box_into_inner)]
+    ///
+    /// let c = Box::new(5);
+    ///
+    /// let five = c.into_inner();
+    /// ```
+    #[unstable(feature = "box_into_inner", issue = "80437")]
+    #[inline]
+    pub fn into_inner(self) -> T {
+        let (ptr, alloc) = Box::into_raw_with_allocator(self);
+        unsafe {
+            let value = ptr::read(ptr);
+            let non_null = NonNull::new_unchecked(ptr);
+            alloc.deallocate(non_null.cast(), Layout::new::<T>());
```

Comment:
Is this deallocating the original memory in heap and returning the value which could be in heap too?

Reply:
The value is already moved out with `ptr::read`.

Reply:
Thanks, I think I get it already. You can just resolve this.